hbase-client/src/main/java/org/apache/hadoop/hbase/client/ClientAsyncPrefetchScanner.java

```diff
@@ -80,13 +77,10 @@ void setPrefetchListener(Consumer<Boolean> prefetchListener) {
 
   @Override
   protected void initCache() {
-    // concurrent cache
-    maxCacheSize = resultSize2CacheSize(maxScannerResultSize);
+    // Override to put a different cache in place of the super's -- a concurrent one.
     cache = new LinkedBlockingQueue<>();
+    maxCacheSize = resultSize2CacheSize(maxScannerResultSize);
     cacheSizeInBytes = new AtomicLong(0);
-    exceptionsQueue = new ConcurrentLinkedQueue<>();
-    prefetcher = new Thread(new PrefetchRunnable());
```

Comment:
We don't need to keep a handle on the spawned thread? It's a daemon so need to explicitly terminate it on shutdown?

Reply:
Nothing was done w/ the reference previously. Yeah, its daemon so should be just dying when process goes away.